cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/tableSegment/RemoveTableSegmentKeyCommand.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.cli.admin.segmentstore.tableSegment;
+
+import com.google.common.base.Charsets;
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.cli.admin.utils.AdminSegmentHelper;
+import io.pravega.client.tables.impl.HashTableIteratorItem;
+import io.pravega.client.tables.impl.TableSegmentKey;
+import io.pravega.shared.protocol.netty.PravegaNodeUri;
+import lombok.Cleanup;
+import org.apache.curator.framework.CuratorFramework;
+
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Executes a RemoveTableSegmentKey request against the chosen Segment Store instance.
+ */
+public class RemoveTableSegmentKeyCommand extends TableSegmentCommand {
+
+    /**
+     * Creates a new instance of RemoveTableSegmentKeyCommand.
+     *
+     * @param args The arguments for the command.
+     */
+    public RemoveTableSegmentKeyCommand(CommandArgs args) {
+        super(args);
+    }
+
+    @Override
+    public void execute() {
+        ensureArgCount(3);
+        ensureSerializersExist();
+
+        final String fullyQualifiedTableSegmentName = getArg(0);
+        final String key = getArg(1);
+        final String segmentStoreHost = getArg(2);
+
+        @Cleanup
+        CuratorFramework zkClient = createZKClient();
+        @Cleanup
+        AdminSegmentHelper adminSegmentHelper = instantiateAdminSegmentHelper(zkClient);
+        CompletableFuture<HashTableIteratorItem<TableSegmentKey>> getKeysReply = adminSegmentHelper.readTableKeys(fullyQualifiedTableSegmentName,
+                new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()), Integer.MAX_VALUE,
+                HashTableIteratorItem.State.EMPTY, super.authHelper.retrieveMasterToken(), 0L);
+
+        TableSegmentKey keysList = getKeysReply.join().getItems()
```

Comment:
It is not clear to me why we need to enumerate the keys first to remove them.
Why can't we remove key directly?
Will remove throw exception if key doesn't exist?

Reply:
No, it does not throw any exception if the key doesn't exist but just for the user to get a correct information that the key user is trying to remove is correct or not.

Reply:
Please double check that no exception is thrown.

Unblocking for now

Reply:
What Sachin said make sense. I think if key doesn't exist then user will get DataNotFoundException.

Reply:
It does not throw any exception if the key doesn't exist.